templates/_worker.js/utils/http.ts

```diff
@@ -16,12 +16,16 @@ export function applyHeaders(
 	const entries =
 		source instanceof Headers ? source.entries() : Object.entries(source);
 	for (const [key, value] of entries) {
-		target.set(
-			key.toLowerCase(),
-			pcreMatch?.match
-				? applyPCREMatches(value, pcreMatch.match, pcreMatch.captureGroupKeys)
-				: value
-		);
+		const lowerKey = key.toLowerCase();
+		const newValue = pcreMatch?.match
+			? applyPCREMatches(value, pcreMatch.match, pcreMatch.captureGroupKeys)
+			: value;
+
+		if (lowerKey === 'set-cookie') {
+			target.append(lowerKey, newValue);
+		} else {
+			target.set(lowerKey, newValue);
```

Comment:
can't we alway use `append`?

Reply:
Afraid not, we seem to end up with multiple `X-Matched-Path` headers when doing so.

Reply:
and naturally I can't see your comment 😅